Kernel/Devices/HID/USB/KeyboardDevice.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022-2023, Liav A. <liavalb@hotmail.co.il>
+ * Copyright (c) 2024, Olekoop <mlglol360xd@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Badge.h>
+#include <AK/CircularQueue.h>
+#include <AK/OwnPtr.h>
+#include <AK/RefPtr.h>
+#include <Kernel/Bus/USB/USBDevice.h>
+#include <Kernel/Bus/USB/USBPipe.h>
+#include <Kernel/Devices/HID/KeyboardDevice.h>
+#include <Kernel/Library/KString.h>
+#include <Kernel/Security/Random.h>
+
+namespace Kernel {
+
+class USBKeyboardDevice final : public KeyboardDevice {
```

Comment:
Fixed that, Now it inherits from `HIDDevice`

Reply:
This is not what I meant - look at how the USB mouse driver has a `USBMouseDevice` which holds a `MouseDevice`:
https://github.com/SerenityOS/serenity/blob/4d5823a5bcc1de687dc7f62ee5bdb35f3e392b02/Kernel/Devices/HID/USB/MouseDevice.h#L34-L37

I'd want the same to happen here.

Reply:
Well, I don't quite understand.
It seems to me like the USB Mouse class inherits from `MouseDevice` and uses itself to process packets
https://github.com/SerenityOS/serenity/blob/4d5823a5bcc1de687dc7f62ee5bdb35f3e392b02/Kernel/Devices/HID/USB/MouseDevice.h#L22-L23
https://github.com/SerenityOS/serenity/blob/4d5823a5bcc1de687dc7f62ee5bdb35f3e392b02/Kernel/Devices/HID/USB/MouseDevice.cpp#L43
While the PS/2 drivers need to have either `KeyboardDevice` or `MouseDevice` passed on
https://github.com/SerenityOS/serenity/blob/4d5823a5bcc1de687dc7f62ee5bdb35f3e392b02/Kernel/Devices/HID/PS2/KeyboardDevice.h#L31
https://github.com/SerenityOS/serenity/blob/4d5823a5bcc1de687dc7f62ee5bdb35f3e392b02/Kernel/Devices/HID/PS2/MouseDevice.h#L30
and use it to process input. They inherit from `SerialIODevice`.
If you want to have just a sepreate `KeyboardDevice`. I have fixed that. If you want something else, tell me.

Reply:
What I wanted is essentially a replicate of the `PS2KeyboardDevice` or `PS2MouseDevice` class.
But now that I look at this again, we should keep it the way it was for now, so you should revert to inherit from `KeyboardDevice` and to not have a separate device, as it doesn't make sense to both inherit and have the same device class as a member.

What we should aim is to have some sort of model like in the PS/2 keyboard and mouse code, but it will take some time to write the proper modifications, so let's not do this for now.

Reply:
Okay, it should be fixed